### ISSUE TITLE
Support for rgba colors

### ISSUE THREAD

**Issue by adius** [mrdoob/three.js#6014]
Shouldn't THREE.color support rgba colors directly?

**Comment by mrdoob** [comment]
I think it used to at some point but `WebGLRenderer` decided to ignore per vertex alpha to save on memory. Whether it was a good decision or not it's up to discussion :)

**Comment by adius** [comment]
mh … rgba colors should at least be handled gracefully. When you specify a rgba color like e.g. 0xffffff55 it will display some seemingly random color because of the bit shifting operations used to get the rgb values from the hex number. There should, however, be a warning or an error, that only rgb values are supported

**Comment by mrdoob** [comment]
Agreed! Would you be up for doing a fix for that?

**Comment by Usnul** [comment]
```javascript
if(hex >= Math.pow(2,24)){
   //shift out alpha
   hex >>= 8;
}
```

**Comment by WestLangley** [comment]
@Usnul That will not work with `0x0000ff55`, for example. Or worse, typos such as `0x0000ff5`.

Documentation should be sufficient, in this case.

If not, then what is our policy on type checking?

**Comment by Usnul** [comment]
you're right, and you're also right that there's no way of figuring out what ff5 is supposed to be, rgb or rgba. That's just a catch for when there are more bits than 24 in a passed number. With strings it's deterministic, but on numbers there's no way to tell how many bits you mean to actually use.
```javascript
var hexDigits = hexString.substring(hexString.indexOf("#")+1);
var rgbDigits = hexDigits.substring(0,6);
```
I'm not really sure what the right strategy would be.
@mrdoob 
where did the memory saving come into play?

also, what about HDR colors? floats for instance. Would it make sense to have colors specified as v4 of floats instead?

**Comment by bhouston** [comment]
Related discussions: https://github.com/mrdoob/three.js/issues/2730  https://github.com/mrdoob/three.js/issues/768

**Comment by metrofun** [comment]
Personally I was misguided by the documentation, that all CSS-style strings are supported.
It worth at least hinting in the documentation for `Color`, that the rgba are not.

**Comment by Mardonis** [comment]
Has there been anymore progression towards this nice possible feature?

**Comment by WestLangley** [comment]
> Has there been anymore progression towards this nice possible feature?

No, but with `ShaderMaterial` and custom attributes, you should be able to implement a work-around for whatever you are trying to do.

**Comment by adius** [comment]
Well there seems to be no easy fix.
We should at least make sure that the documentation is sufficient.
…but I think @Usnul 's idea is better than nothing and should be considered!

**Comment by dubejf** [comment]
Fixed in #6898 - rgba colors are handled more gracefully. The alpha component is ignored because `THREE.Color` does not support them, but the color components are read.
